### DISCUSSION TITLE
Receiving 'Not connected, trying to connect!' continually (Sync)

### DISCUSSION THREAD

**Discussion by MightiestSleep** [pymodbus-dev/pymodbus#2495] [Help] [answered]
Hi!
I am currently working on a scanning utility with PyModbus. It seems that whenever I try to have a persistent connection, I receive a log warning 'Not connected, trying to connect!'. I converted my file over to the Async client, and did not receive this log. Additionally, in my conversion, I did my best to keep it as similar as possible. I have a simple, sandboxed version of my utility for the sake of simple debugging:
```
`''' Future annotations '''
from __future__ import annotations
from abc import ABC, abstractmethod

from typing import Dict, List, Union, Optional, Literal
import time as time
from datetime import datetime
from dataclasses import dataclass, field
import yaml
from pymodbus.client import ModbusTcpClient, AsyncModbusSerialClient
import asyncio
from pymodbus.client.mixin import ModbusClientMixin
from pymodbus.exceptions import ConnectionException, ModbusException
from pymodbus.payload import BinaryPayloadBuilder, BinaryPayloadDecoder
from pymodbus.constants import Endian


class ModbusRead:
    def __init__(self, client):
        self.client: ModbusTcpClient = client
        self.scanning: bool = False
        self.client.connect()
    

    def perform_scan(self):
        print('Starting scan')
        while self.scanning:
            self.read_modbus()
            time.sleep(1)

    def start_scan(self):
        self.scanning = True
        self.perform_scan()

    def stop_scan(self):
        self.scanning = False
        print('Scan stopped')

    def read_modbus(self):
        # Ensure client is connected before trying to read

        #self.client.connect()  # Reconnect if necessary
        # Perform the read operation
        start_time = time.time()
        response = self.client.read_holding_registers(address=42772, count=1)
        end_time = time.time()
        execution_time = end_time - start_time
        print(f"execution time: {execution_time} seconds for holding registers")
        if response.isError():
            print("Error reading modbus:", response)
        else:
            print("Response Registers:", response.registers)

def main():
    # Initialize the ModbusRead object with a connected AsyncModbusTcpClient
    modbus_device = ModbusRead(client=ModbusTcpClient(host='192.168.5.198'))
    # Start scanning asynchronously
    modbus_device.start_scan()
    
    # Simulate some other tasks or delays
    time.sleep(5)
    
    print('TEST FUNCTION')

    print("Scan stopped:", modbus_device.scanning)

if __name__ == "__main__":
    main()
```
When I log this, I get the following:
`Not connected, trying to connect!
execution time: 0.001957416534423828 seconds for holding registers
Response Registers: [3]`
I have attempted this with a few simulated Modbus devices, as well as a standard Modbus controller, and receive the same thing. It doesn't seem to impact my ability to receive data, and I am content with the ~1ms response time.
Any help would be greatly appreciated!

**Comment by janiversen** [comment]
You do not write which version of pymodbus you are using, but I assume it is 3.8.0

In case of v3.8.0 this is a harmless log message, which have been removed (available on dev).

**Reply by MightiestSleep** [reply]
> You do not write which version of pymodbus you are using, but I assume it is 3.8.0
> 
> In case of v3.8.0 this is a harmless log message, which have been removed (available on dev).

My apologies, I AM using 3.8.0. I am glad to hear that it's benign. Out of curiosity, do you have any insights as to why this is happening? I scrolled through the source code a little but admittedly lack some context.

Thank you for your reply!

**Comment by janiversen** [comment, marked as answer]
Sorry my bad, yes the code works with 3..8.

It is solved on dev, look there.
